CHANGELOG.md

```diff
@@ -24,6 +24,13 @@ Changelog
 
 * User context is more reliable not cache when the hash mismatches. (E.g. after
   login/logout.)
+  
+* The ContextInvalidationLogoutHandler has been deprecated in favor of the 
+  ContextInvalidationSessionLogoutHandler.  The original handler was called 
+  after the invalidation of the session so did not correctly invalidate the 
```

Comment:
... session so did not correctly invalidate the session ID in cache... => session, and thus did not invalidate the session it should have but a newly created one.